contracts/main/voice/src/ibc.rs

```diff
@@ -84,7 +84,7 @@ pub fn ibc_packet_receive(
     let connection_id = CHANNEL_TO_CONNECTION
         .load(deps.storage, msg.packet.dest.channel_id.clone())
         .expect("handshake sets mapping");
-    Ok(IbcReceiveResponse::default()
+    Ok(IbcReceiveResponse::without_ack()
```

Comment:
This is a bit scary. Are you sure you don't want an acknowledgement here? To get back the old behaviour use


```suggestion
    Ok(IbcReceiveResponse::new(b"")
```

Reply:
Thank you! Didn't know it's a different behavior

Reply:
`::new(b"")` writes an empty acknowledgement. No idea if this is a wise thing to do but that is what the old code did.

CosmWasm 2 allows you to not write an acknowledgement at all with `::without_ack()`. This is for protocols that don't have acknowledgements (uncommen edge case) or use asynchonous acknowledgements (also advanced stuff).

Reply:
How can the acknowledgment be written at a later point? Is there an example of how to use this?

In polytone's case it's a bit confusing because the receiver of the IBC packet wants to send back the result of the execution. Shouldn't we write that acknowledgment at the end of the last action in that case?

Reply:
CosmWasm 2.1 adds `IbcMsg::WriteAcknowledgement` which allows you to write the acknowledgement in a different transaction or block. There is a bit of example code in the cosmwasm repo.

Reply:
I don't know how polytone works, so I cannot really recommend what is the best way to do it in your cae

Reply:
The idea is that this function leaves no ack (or an empty one) and lets the reply modify that ack with fresh data. 
So both would be ok if I understand correctly ? As what is registered there is overwritten later

Reply:
You should never try to override an ack. I guess this is even explicitly blocked. But what you can do indeed for all CosmWasm 2.1+ chains is not wriing an ack here (`:without_ack()`) and then write it in the reply with `IbcMsg::WriteAcknowledgement`.

Reply:
Please note that the chain needs wasmd 0.52 in order to fully support `:without_ack()` due to this bug. https://github.com/CosmWasm/wasmd/issues/1891. I.e. current Neutron for example does not support it yet.